### PR TITLE
fix: blockName regexp incorrect

### PR DESCRIPTION
**Pull request by chilingling**
English | [简体中文](https://github.com/opentiny/tiny-engine/blob/develop/.github/PULL_REQUEST_TEMPLATE/PULL_REQUEST_TEMPLATE.zh-CN.md)

# PR

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our [Commit Message Guidelines](https://github.com/opentiny/tiny-engine/blob/develop/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Built its own designer, fully self-validated

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

### What is the current behavior?

原来的正则：`/^([A-Z][A-Za-z0-9]{2,})*?([A-Z][A-Za-z0-9]{2,})*?$/`

该正则会导致问题

- 只有一个单词也能通过校验，比如：`Abb` 也能通过校验

Issue Number: N/A

### What is the new behavior?

新正则：`/^[A-Z][A-Za-z0-9]*?[A-Z][A-Za-z0-9]*$/`

与后端对齐，必须两个单词以上。即 `Abb` 无法通过校验。

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Summary by CodeRabbit

- **New Features**
	- Updated validation criteria for block names, now requiring that block names start with an uppercase letter followed by at least one lowercase letter or digit, and must contain at least two such segments.

- **Bug Fixes**
	- Improved accuracy of block name validation to enhance user input reliability.

### CODE REVIEW

**Review by yy-wow — DISMISSED** [review]

File: packages/controller/js/verification.js [hunk 1]
```diff
@@ -1,20 +1,20 @@
...
+ * A PARTICULAR PURPOSE. SEE THE APPLICABLE LICENSES FOR MORE DETAILS.
+ *
+ */
 
 export const REGEXP_EVENT_NAME = /^[a-z]+([A-Z][a-z]*)*$/
 
 export const verifyEventName = (name) => REGEXP_EVENT_NAME.test(name)
 
-export const REGEXP_BLOCK_NAME = /^([A-Z][A-Za-z0-9]{2,})*?([A-Z][A-Za-z0-9]{2,})*?$/
+export const REGEXP_BLOCK_NAME = /^([A-Z][A-Za-z0-9]{1,}){2,}$/
```
Comment (rhlin) [inline packages/controller/js/verification.js]: 该正则双层可变长度字符串且有重合部分可能会造成正则回溯性灾难导致性能问题， 且不支持ColumnA, DataB这样的单词，建议替换一下

Comment (chilingling) [reply packages/controller/js/verification.js]: Done

**Review by rhlin — DISMISSED** [review]
LGTM now

File: packages/controller/js/verification.js [hunk 2]
```diff
@@ -1,20 +1,20 @@
...
+ * A PARTICULAR PURPOSE. SEE THE APPLICABLE LICENSES FOR MORE DETAILS.
+ *
+ */
 
 export const REGEXP_EVENT_NAME = /^[a-z]+([A-Z][a-z]*)*$/
 
 export const verifyEventName = (name) => REGEXP_EVENT_NAME.test(name)
 
-export const REGEXP_BLOCK_NAME = /^([A-Z][A-Za-z0-9]{2,})*?([A-Z][A-Za-z0-9]{2,})*?$/
+export const REGEXP_BLOCK_NAME = /^[A-Z][A-Za-z0-9]*?[A-Z][A-Za-z0-9]*$/
```
Comment (hexqi) [inline packages/controller/js/verification.js:17]: 第二个A-Z看起来可以省略，省略后会更清晰一点，/^[A-Z][a-z0-9]*?[A-Z][A-Za-z0-9]*$/

Comment (hexqi) [inline packages/controller/js/verification.js]: 写成这样感觉更简短、容易理解，`/^([A-Z][a-z0-9]*){2,}$/`,  `([A-Z][a-z0-9]*)`表示一个单词，匹配两次及以上

Comment (hexqi) [reply packages/controller/js/verification.js]: 或者这样：/^[A-Z][a-z0-9]*?[A-Z][A-Za-z0-9]*$/

**Review by hexqi — APPROVED** [review]

**Pull request merged**